packages/components/icon/src/generateIcon.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { forwardRef } from 'react';
+import type { ReactElement } from 'react';
+
+import { Icon } from './Icon';
+import type { IconProps } from './Icon';
+
+export type GeneratedIconProps = Omit<
```

Comment:
Should this be `IconInternalProps` same pattern as https://github.com/contentful/forma-36/pull/745/files#diff-bbad09ab711a522cbdab27581c9bb3f575da40589810fbeab1754b2bd320d2cfR25 ?

Reply:
We want all the other stuff that gets added to `IconProps` like `className` and `testId`.

The difference between this case and the linked file is that the linked component extends the internal props and then uses `PolymorphicComponentProps` on its own props to add the shared props, , but in this case we're not dealing with a component with props, so I would argue that it's right to `Omit` types we don't allow instead of using `PolymorphicComponentProps`.